cassovary-core/src/main/scala/com/twitter/cassovary/util/io/AdjacencyListGraphReader.scala

```diff
@@ -111,10 +110,7 @@ class AdjacencyListGraphReader[T] (
             i += 1
           }
 
-          holder.id = internalNodeId
-          holder.edges = outEdgesArr
-          holder.maxId = newMaxId
-          holder
+          NodeIdEdgesMaxId(internalNodeId, outEdgesArr, newMaxId)
```

Comment:
We might prefer not creating the object (I'll let Pankaj answer that) but I don't think it should matter whether the graph is immutable or not. The question is whether the NodeIdEdgesMaxId returned by each iteration should be immutable or not.